rpc/jsonrpc/client/decode.go

```diff
@@ -40,62 +39,32 @@ func unmarshalResponseBytesArray(responseBytes []byte, expectedIDs []rpctypes.JS
 	}
 
 	// Intersect IDs from responses with expectedIDs.
-	ids := make([]rpctypes.JSONRPCIntID, len(responses))
-	var ok bool
+	ids := make([]string, len(responses))
 	for i, resp := range responses {
-		ids[i], ok = resp.ID.(rpctypes.JSONRPCIntID)
-		if !ok {
-			return fmt.Errorf("expected JSONRPCIntID, got %T", resp.ID)
-		}
+		ids[i] = resp.ID()
 	}
 	if err := validateResponseIDs(ids, expectedIDs); err != nil {
 		return fmt.Errorf("wrong IDs: %w", err)
 	}
 
-	for i := 0; i < len(responses); i++ {
-		if err := json.Unmarshal(responses[i].Result, results[i]); err != nil {
-			return fmt.Errorf("error unmarshaling #%d result: %w", i, err)
+	for i, resp := range responses {
+		if err := json.Unmarshal(resp.Result, results[i]); err != nil {
+			return fmt.Errorf("unmarshaling result %d: %w", i, err)
 		}
 	}
 	return nil
 }
 
-func validateResponseIDs(ids, expectedIDs []rpctypes.JSONRPCIntID) error {
-	m := make(map[rpctypes.JSONRPCIntID]bool, len(expectedIDs))
-	for _, expectedID := range expectedIDs {
-		m[expectedID] = true
+func validateResponseIDs(ids, expectedIDs []string) error {
+	m := make(map[string]bool, len(expectedIDs))
```

Comment:
Isn't `map[string]struct{}` usually the set type?

Reply:
Yeah. It doesn't really matter for a small, short-lived map, though. The main reason to care is if you have a large map that hangs around, since it saves storage for the values. Here it's just an ergonomics thing.

(Nonetheless, it was an easy change, so I did it)